consensus/istanbul/logger.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 The celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package istanbul
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"math/big"
+)
+
+type istLogger struct {
+	logger log.Logger
+	round  func() *big.Int
+}
+
+// NewIstLogger creates an Istanbul Logger with custom logic for exposing logs
+func NewIstLogger(fn func() *big.Int, ctx ...interface{}) log.Logger {
+	return &istLogger{logger: log.New(ctx...), round: fn}
+}
+
+func (l *istLogger) New(ctx ...interface{}) log.Logger {
+	childLogger := l.logger.New(ctx...)
+	return &istLogger{logger: childLogger, round: l.round}
+}
+
+func (l *istLogger) Trace(msg string, ctx ...interface{}) {
+	// If the current round > 1, then upgrade this message to Info
+	if l.round().Cmp(common.Big1) > 0 {
```

Comment:
could you add a check to gracefully handle the case where `l.round` is `nil`

Reply:
Done, good call